### PR TITLE
Sweble2

### PR DESCRIPTION
**Pull request by aditya-nambiar**

### CODE REVIEW

File: core/src/main/scala/org/dbpedia/extraction/wikiparser/NodeUtil.scala [hunk 1]
```diff
@@ -210,4 +213,32 @@ object NodeUtil
...
+
+    def notEmpty(node : Node) : Boolean = {
+
+        if(!node.isInstanceOf[TextNode]){
+            return true;
+        }
+
+        var content = node.asInstanceOf[TextNode].text.trim
+
+        if (content == "" || content == "\n"){
```
Comment (jimkont) [inline core/src/main/scala/org/dbpedia/extraction/wikiparser/NodeUtil.scala]: `trim()` removes any whitespace like `\n`
something like `return ! node.asInstanceOf[TextNode].text.trim.empty` should work

File: core/src/main/scala/org/dbpedia/extraction/wikiparser/NodeUtil.scala [hunk 2]
```diff
@@ -210,4 +213,32 @@ object NodeUtil
 
         splitNodes.reverse
     }
+
+    def removeEmptyTextNode(list : List[Node]) : List[Node] = {
+        var tempList = ListBuffer[Node]()
```
Comment (jimkont) [inline core/src/main/scala/org/dbpedia/extraction/wikiparser/NodeUtil.scala]: could be something like `return list.filter(x => notEmpty(x))`

File: core/src/main/scala/org/dbpedia/extraction/wikiparser/NodeUtil.scala [hunk 3]
```diff
@@ -210,4 +213,32 @@ object NodeUtil
...
+
+        for( node <- list){
+            if(notEmpty(node)){
+                tempList += node
+            }
+        }
+        tempList.toList
+    }
+
+    def notEmpty(node : Node) : Boolean = {
```
Comment (jimkont) [inline core/src/main/scala/org/dbpedia/extraction/wikiparser/NodeUtil.scala]: better rename to `isEmptyTextNode`

File: core/src/main/scala/org/dbpedia/extraction/wikiparser/NodeUtil.scala [hunk 4]
```diff
@@ -210,4 +213,32 @@ object NodeUtil
 
         splitNodes.reverse
     }
+
+    def removeEmptyTextNode(list : List[Node]) : List[Node] = {
```
Comment (jimkont) [inline core/src/main/scala/org/dbpedia/extraction/wikiparser/NodeUtil.scala]: `remove` sounds dangerous and not clear if the nodes are deleted or not :)
`copy excluding` or `copy without` sounds safer :)

**Comment by jimkont** [comment]
Hey @aditya-nambiar , this is good to merge (Y) but before I do can you please remove `core/src/test/scala/org/dbpedia/extraction/.DS_Store` from the git history and update your branch?

**Comment by jimkont** [comment]
Thanks @aditya-nambiar  !

**Comment by aditya-nambiar** [comment]
Welcome :) Sorry was a bit busy earlier. Also thanks for the awesome evaluation!

**Pull request merged**